packages/connect-query/src/use-query.ts

```diff
@@ -59,18 +59,10 @@ export function useQuery<
     transport: transport ?? transportFromCtx,
     callOptions,
   });
-  const { enabled: baseEnabled, ...baseRest } = baseOptions;
-  const tsOpts = {
+  return tsUseQuery({
     ...queryOptions,
-    ...baseRest,
-  };
-  // The query cannot be enabled if the base options are disabled, regardless of
-  // incoming query options.
-  const enabled = baseEnabled ?? queryOptions.enabled;
-  if (enabled !== undefined) {
-    tsOpts.enabled = enabled;
-  }
-  return tsUseQuery(tsOpts);
+    ...baseOptions,
+  });
```

Comment:
Seems easier to read to me.

I wonder if we should change the order here, and let user-provided options override our options. This would provide an escape hatch for things like #418.

TanStack Query let's users provide a `queryHash` even though it's usually computed from the key. Seems fine to me to let users provide a `queryFn`, `queryKey`, or `structuralSharing` in our hooks, even if the typical use case is to compute them.

Reply:
I feel like i'd want a more tailored way to customize generated keys rather than just overriding. If our intention is to build a typesafe version of react-query, then allowing customization of queryFn could be a problem, not to mention what happens if you have different call sites using different versions of that queryFn. Worth thinking through but I do like the cleanup anyways.

Reply:
Let's discuss in https://github.com/connectrpc/connect-query-es/pull/428